Library/Homebrew/dependencies_helpers.rb

```diff
@@ -23,9 +19,35 @@ def args_includes_ignores(args)
     [includes, ignores]
   end
 
-  def recursive_includes(klass, root_dependent, includes, ignores)
-    raise ArgumentError, "Invalid class argument: #{klass}" if klass != Dependency && klass != Requirement
+  sig {
+    params(root_dependent: T.any(Formula, CaskDependent), includes: T::Array[Symbol], ignores: T::Array[Symbol])
+      .returns(T::Array[Dependency])
+  }
+  def recursive_dep_includes(root_dependent, includes, ignores)
+    # The use of T.unsafe is recommended by the Sorbet docs:
+    #   https://sorbet.org/docs/overloads#multiple-methods-but-sharing-a-common-implementation
+    T.unsafe(recursive_includes(Dependency, root_dependent, includes, ignores))
+  end
 
+  sig {
+    params(root_dependent: T.any(Formula, CaskDependent), includes: T::Array[Symbol], ignores: T::Array[Symbol])
+      .returns(Requirements)
+  }
+  def recursive_req_includes(root_dependent, includes, ignores)
+    # The use of T.unsafe is recommended by the Sorbet docs:
+    #   https://sorbet.org/docs/overloads#multiple-methods-but-sharing-a-common-implementation
+    T.unsafe(recursive_includes(Requirement, root_dependent, includes, ignores))
+  end
+
+  sig {
+    params(
+      klass:          T.any(T.class_of(Dependency), T.class_of(Requirement)),
+      root_dependent: T.any(Formula, CaskDependent),
+      includes:       T::Array[Symbol],
+      ignores:        T::Array[Symbol],
+    ).returns(T.any(T::Array[Dependency], Requirements))
+  }
+  def recursive_includes(klass, root_dependent, includes, ignores)
```

Comment:
I added the two wrapper methods above, to avoid a union return type that would have to be addressed at the caller.